src/main/webapp/app/entities/submission.model.ts

```diff
@@ -40,3 +40,58 @@ export abstract class Submission implements BaseEntity {
         this.submitted = false; // default value
     }
 }
+
+/**
+ * Used to access the correct submissions result
+ *
+ * @param submission
+ */
+export function getLatestSubmissionResult(submission: Submission | undefined): Result | undefined {
```

Comment:
I am not sure if it would be better to not export this function and rename it to `getLatestResult` then you can call this directly from a submission, e.g. `const result = this.submission.getLatestResult()`

Reply:
This does not work in typescript because there is no actual object submission in runtime. You cant really work in an object oriented way